src/pages/core/specification/instantiate2-algo.mdx

```diff
@@ -0,0 +1,46 @@
+---
+tags: ["core", "specification"]
+---
+
+import { Callout } from "nextra/components";
+
+# Instantiate2 algorithm
+
+With the instantiate2 algorithm you can create a contract address in a
+predictable and deterministic way. The underlying algorithm is rather simple.
+
+We use SHA-256 as the underlying hashing algorithm.
+
+You need to provide the following inputs:
+
+- Checksum: This is the checksum of the contract code (the Wasm module, for
+  example). This _has_ to be a SHA-256 hash
+- Creator: This is the canonicalized address of, for example, the contract
```

Comment:
"contract instantiating the contract"? That's surely possible (and common in a "factory->suite of contracts" style app), but I think it's simpler to say "user instantiating the contract"